src/osd/ReplicatedPG.cc

```diff
@@ -10462,6 +10423,11 @@ void ReplicatedPG::hit_set_trim(RepGather *repop, unsigned max)
   for (unsigned num = updated_hit_set_hist.history.size(); num > max; --num) {
     list<pg_hit_set_info_t>::iterator p = updated_hit_set_hist.history.begin();
     assert(p != updated_hit_set_hist.history.end());
+    if (!p->using_gmt) {
```

Comment:
i agree. probably we can have a release notes instructing user to disable hit set temporarily before upgrade: https://github.com/ceph/ceph/pull/4998#issuecomment-113038846